conan/api/subapi/list.py

```diff
@@ -84,9 +84,23 @@ def filter_packages_configurations(pkg_configurations, query):
         """
         return filter_packages(query, pkg_configurations)
 
-    def select(self, pattern, package_query=None, remote=None):
-        if package_query and pattern.package_id and "*" not in pattern.package_id:
-            raise ConanException("Cannot specify '-p' package queries, "
+    @staticmethod
+    def filter_packages_profile(pkg_configurations, profile):
+        profile_settings = profile.settings
+
+        def _matches(d):  # Only implements settings matching at this moment
+            settings = d.get("settings", {})
+            for k, v in settings.items():
+                profile_value = profile_settings.get(k)
+                if profile_value is not None and profile_value != v:
+                    return False
+            return True
+
+        return {pref: data for pref, data in pkg_configurations.items() if _matches(data)}
```

Comment:
Would it be interesting to have implemented the `__eq__` operator on the Settings side? I see that `SettingsItem` already has it so I think it would make sense.